app/Exceptions/Handlers/NoEncryptionKey.php

```diff
@@ -26,6 +26,9 @@ public function check(HttpException $e): bool
 			if ($e instanceof MissingAppKeyException) {
 				return true;
 			}
+			if ($e->getMessage() === 'No application encryption key has been specified.') {
```

Comment:
Laravel 8.

Previously, Laravel was directly throwing the Exception, now if it is not an HttpException, the handler converts it into one, keeps the message AND DROP the previous exception.

Why did this behavior change on Laravel side? I don't know and I don't really care TBH.

Reply:
> Previously, Laravel was directly throwing the `MissingAppKeyException`, now if it is not an HttpException, the handler converts it into one

Wait! Are you telling me that with the transition from Laravel 8 to 9 "something" inside the framework converts exceptions into `HttpException`? If yes, then this problem needs a much more thorough investigation, because there is much more code that checks exceptions types and relies on it. This would be a fundamental change and must be treated as such. This would also imply that the error reporting towards the JSON client might not work anymore the way as it has been used to. This cannot be brushed aside by a simple patch.

> Why did this behavior change on Laravel side? I don't know and I don't really care TBH.

That is exactly what I suspected and how that PR felt like.

Reply:
This IS STILL LARAVEL 8 !!! Nothing related to Laravel 9.

Reply:
I fixed it myself in https://github.com/LycheeOrg/Lychee/pull/1495/commits/27ad036e6a4496fec6849b2179809ab12bea4186. The actual fix is a one-liner. I also created a PR for the Laravel framework: https://github.com/laravel/framework/pull/43991.

The remaining changes are only a lot of additional comments. At some day I will completely replace this ugly class.